src/main/Port.Adapter/UI/Views/Blazor.Common/TreeView.razor

```diff
@@ -316,6 +319,147 @@
+    {
+        try
+        {
+            if (this.Children.Contains(child))
+            {
+                //Remove highlight from current node
+                if (!Recursive)
+                {
+                    await JSRuntime.InvokeVoidAsync("hover", "g", child.Neuron.Id, false, "highlightNode");
+                    //await OnUnhover.InvokeAsync(child);
+                }
+
+                //Propagate Unhover to children if expanded
+                if(child.ExpansionState == ExpansionState.Expanded && child.Children != null && child.Children.Any())
+                {
+                    foreach(var childNode in child.Children )
+                    {
+                        //Remove highlight from each child node
+                        if (childNode.Neuron.Id == Id)
+                            continue;
+                        await JSRuntime.InvokeVoidAsync("hover", "g", childNode.Neuron.Id, false, "highlightChildNode");
+
+                        //Recursively remove highlight from children of children if they're expanded
+                        if(childNode.ExpansionState == ExpansionState.Expanded && childNode.Children != null && childNode.Children.Any())
+                        {
+                            await Unhover(childNode, true, childNode.Neuron.Id);
+                        }
+                    }
+                }
+            }
+            else
+            {
+                var newChild = this.Children.FirstOrDefault(x => x.Children.Contains(child));
+                if (!Recursive)
+                {
+                    await JSRuntime.InvokeVoidAsync("hover", "g", newChild.Neuron.Id, false, "highlightNode");
+                    //await OnUnhover.InvokeAsync(newChild);
+                }
+
+                if (child.ExpansionState == ExpansionState.Expanded && child.Children != null && child.Children.Any())
+                {
+                    foreach (var childNode in child.Children)
+                    {
+                        //Remove highlight from each child node
+                        if (childNode.Neuron.Id == Id)
+                            continue;
+                        await JSRuntime.InvokeVoidAsync("hover", "g", childNode.Neuron.Id, false, "highlightChildNode");
+
+                        //Recursively remove highlight from children of children if they're expanded
+                        if (childNode.ExpansionState == ExpansionState.Expanded && childNode.Children != null && childNode.Children.Any())
+                        {
+                            await Unhover(childNode, true, childNode.Neuron.Id);
+                        }
+                    }
+                }
+            }
+            return "";
+        }
+        catch (Exception ex)
+        {
+            var x = ex.Message;
+            return x;
+        }
+    }
+
+    private bool HoverCore(TreeNeuronViewModel childNode,string Id)
```

Comment:
HoverCore should consist of the code inside Hover and Unhover, the line below should be a delegate parameter.